src/adapters/class-jw-player.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * WP Video Sync: JW Player Adapter.
+ *
+ * @package wp-video-sync
+ */
+
+namespace Alley\WP\WP_Video_Sync\Adapters;
+
+use Alley\WP\WP_Video_Sync\API\JW_Player_API;
+use Alley\WP\WP_Video_Sync\Interfaces\Adapter;
+use DateTimeImmutable;
+use stdClass;
+
+/**
+ * JW Player Adapter.
+ */
+class JW_Player implements Adapter {
+
+	/**
+	 * The date of the last modification to the last batch of videos.
+	 *
+	 * @var ?DateTimeImmutable
+	 */
+	private ?DateTimeImmutable $last_modified_date;
+
+	/**
+	 * The JW Player API.
+	 *
+	 * @var JW_Player_API
+	 */
+	public JW_Player_API $jw_player_api;
+
+	/**
+	 * The start date if the `last_modified_date` does not exist.
+	 *
+	 * @var string
+	 */
+	public string $origin_modified_date;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param JW_Player_API $api Instance of the JW Player API object.
+	 * @param string        $origin_modified_date The date of the used if the `last_modified_date` does not exist.
+	 */
+	public function __construct( JW_Player_API $api, string $origin_modified_date = '' ) {
+		$this->jw_player_api        = $api;
+		$this->origin_modified_date = ! empty( $origin_modified_date ) ? $origin_modified_date : date( 'Y-m-d' );
+	}
+
+	/**
+	 * Fetches the date of the last modification to the last batch of videos.
+	 *
+	 * @return ?DateTimeImmutable
+	 */
+	public function get_last_modified_date(): ?DateTimeImmutable {
+		return $this->last_modified_date ?? DateTimeImmutable::createFromFormat('Y-m-d', $this->origin_modified_date );
```

Comment:
With advice above this becomes a simple return of the class property.

Reply:
I've updated this - is this a case where these jwp classes should extend a parent class?

Reply:
It could, yeah, especially if your API class and the JW Player 7 class are both handling data in the same format (which they should be - the JW Player 7 class is using the JW Player 7 plugin's API helper methods and credentials and otherwise returning what it gets)